lib/ansi/ansi.go

```diff
@@ -0,0 +1,11 @@
+// Package ansi provides facilities to deal with ANSI codes
+package ansi
+
+import (
+	"fmt"
+)
+
+// WriteTerminalTitle writes a string to the terminal title
+func WriteTerminalTitle(title string) {
```

Comment:
This should go in  

- https://github.com/rclone/rclone/blob/master/lib/terminal/terminal_normal.go

With a null implementation here

- https://github.com/rclone/rclone/blob/master/lib/terminal/terminal_unsupported.go 

I'd probably put the constants in

- https://github.com/rclone/rclone/blob/master/lib/terminal/terminal.go

And I'd put all the `lib/terminal` changes into a seperate commit called "lib/terminal: add WriteTerminalTitle" or similar.